### PR TITLE
Minor quality of life suggestions

### PR DESCRIPTION
**Pull request by simon-curtis**
I have: 

- Added a RawValue property to the Ok result, in case someone wanted to cache the result or do custom deserialisation.
- Added an options builder delegate extension method `.AddSurrealDb(o => o.WithDatabase(...))`, it's the type I personally reach for the most when configuring services.

### CODE REVIEW

**Review by Odonno — CHANGES_REQUESTED** [review]
Looks nice. Some minor changes requested.

**Comment by simon-curtis** [comment]
Regards the RawValue and Deserialising... it might make sense for us to just expose the JsonElement with a private setter?

File: SurrealDb.Net/Models/Response/SurrealDbResult.Ok.cs [hunk 1]
```diff
@@ -22,6 +22,11 @@ public sealed class SurrealDbOkResult : ISurrealDbResult
 
     public bool IsOk => true;
 
+    /// <summary>
+    /// Displays the raw JSON value of the query result.
+    /// </summary>
+    public string RawValue => _value.GetRawText();
+
```
[the remark this answers is not in the document]
Comment (simon-curtis) [reply SurrealDb.Net/Models/Response/SurrealDbResult.Ok.cs:29]: We already have an internal `IEnumerable<T> DeserializeEnumerable<T>()` signature

[the remark this answers is not in the document]
Comment (simon-curtis) [reply SurrealDb.Net/Models/Response/SurrealDbResult.Ok.cs:29]: We already had `internal IEnumerable<T> DeserializeEnumerable<T>()` but I have made a new public method with the `GetValues` name to keep it inline with GetValue

File: SurrealDb.Net/Models/Response/SurrealDbResult.Ok.cs [hunk 2]
```diff
@@ -22,6 +22,11 @@ public sealed class SurrealDbOkResult : ISurrealDbResult
 
     public bool IsOk => true;
 
+    /// <summary>
+    /// Displays the raw JSON value of the query result.
+    /// </summary>
+    public string RawValue => _value.GetRawText();
```
[the remark this answers is not in the document]
Comment (simon-curtis) [reply SurrealDb.Net/Models/Response/SurrealDbResult.Ok.cs:28]: See my comment reg. exposing _value as a get-only property instead

[the remark this answers is not in the document]
Comment (simon-curtis) [reply SurrealDb.Net/Models/Response/SurrealDbResult.Ok.cs:28]: See my comment reg. exposing _value as a get-only property instead

**Comment by simon-curtis** [comment]
I've made the changes requested. Also I have added a check in the `SurrealDbOkResult.DeserializeEnumerable<T>()` to ensure that the result of the query was an array. I was going to just yield the object and then return early but if someone is calling `DeserializeEnumerable` (or `GetValues`) then I would assume the behaivour they were after was an array. I.e. if they have queried `SELECT * FROM ONLY person` I would not expect them to call an array method.

**Review by Odonno — APPROVED** [review]
[image](https://media.tenor.com/JMr_dlDTHJIAAAAC/theboys-perfect.gif)

**Comment by kearfy** [comment]
Hey @simon-curtis, formatting seems to fail. Could you check that out? Thanks!

**Comment by Odonno** [comment]
Indeed, this project is using https://csharpier.com/

You can run it locally in the cli by installing the dotnet tool on your machine. Or automatically in your favorite IDE as an extension. There is one for Rider.

I will update the readme for newcomers.

**Comment by simon-curtis** [comment]
I've formatted but it hasn't made any changes?

**Comment by simon-curtis** [comment]
All done! Codespaces had checkout the wrong branch for some reason. Yes the space after the the property is gone now

**Pull request merged**